### PR TITLE
Add parsing for non 1-minute data to UO SRML parser

### LINKED ISSUE #709: Erroneous timestamp parsing in UO SRML reader
**Describe the bug**
The parser for UO SRML data does not parse times correctly for 5 minute, 15 minute and hourly data. This is caused by the logic for handling the time format in SRML files always assuming one minute data.

**To Reproduce**
```
import pvlib

# requests a 15 minute file
pvlib.iotools.read_srml_month_from_solardat('TW', 2019, 5, 'RQ')
```
Returns:
```
                          ghi_1  ghi_1_flag  dhi_1  dhi_1_flag  temp_air_1  temp_air_1_flag        ...   
2019-05-01 00:14:00-08:00    0.0          11    0.0          11         1.0               11        ...               
2019-05-01 00:29:00-08:00    0.0          11    0.0          11         0.3               11        ... 
2019-05-01 00:44:00-08:00    0.0          11    0.0          11        -0.9               11        ... 
```

**Expected behavior**
Timestamps should be parsed to align with each hour and to label the beginning of each time interval as noted in the function documentation. e.g. The result of the example above should be:
```
                          ghi_1  ghi_1_flag  dhi_1  dhi_1_flag  temp_air_1  temp_air_1_flag        ...   
2019-05-01 00:00:00-08:00    0.0          11    0.0          11         1.0               11        ...               
2019-05-01 00:15:00-08:00    0.0          11    0.0          11         0.3               11        ... 
2019-05-01 00:30:00-08:00    0.0          11    0.0          11        -0.9               11        ... 
```

**Versions:**
 - ``pvlib.__version__``: 0.6.1
 - ``pandas.__version__``:  0.24.2
 - python: 3.7

### PR DESCRIPTION
**Pull request by lboeman**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python! You may delete all of these instructions except for the list below.

You may submit a pull request with your code at any stage of completion.

The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items below:

 - [x] Closes  #709 
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [ ] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [ ] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.

### CODE REVIEW

File: pvlib/iotools/srml.py [hunk 1]
```diff
@@ -42,11 +42,11 @@ def read_srml(filename):
...
     Notes
     -----
-    The time index is shifted back one minute to account for 2400 hours,
-    and to avoid time parsing errors on leap years. The returned data
-    values should be understood to occur during the interval from the
-    time of the row until the time of the next row. This is consistent
-    with pandas' default labeling behavior.
+    The time index is shifted back by one interval to account for the
+    daily endtime of 2400, and to avoid time parsing errors on leap
+    years. The returned data values should be understood to occur
```
Comment (cwhanse) [inline pvlib/iotools/srml.py]: `are labeled by the left endpoint of interval, and should be understood...`

File: pvlib/iotools/srml.py [hunk 2]
```diff
@@ -134,11 +134,17 @@ def format_index(df):
...
     # Times are expressed as integers from 1-2400, we convert to 0-2359 by
-    # subracting one and then correcting the minutes at each former hour.
-    df_time = df[df.columns[1]] - 1
-    fifty_nines = df_time % 100 == 99
-    times = df_time.where(~fifty_nines, df_time - 40)
-
+    # subracting the length of one interval and then correcting the times
+    # at each former hour. interval_length is determined by taking the
+    # difference of the first two rows of the time column.
+    interval_length = int(df[df.columns[1]][:2].diff()[1])
```
Comment (cwhanse) [inline pvlib/iotools/srml.py]: This line would be easier to read as, e.g., `interval_length = df[df.columns[0]][1] - df[df.columns[0]][0]`

File: pvlib/iotools/srml.py [hunk 3]
```diff
@@ -134,11 +135,17 @@ def format_index(df):
     year = int(df.columns[1])
     df_doy = df[df.columns[0]]
     # Times are expressed as integers from 1-2400, we convert to 0-2359 by
-    # subracting one and then correcting the minutes at each former hour.
-    df_time = df[df.columns[1]] - 1
-    fifty_nines = df_time % 100 == 99
-    times = df_time.where(~fifty_nines, df_time - 40)
-
+    # subracting the length of one interval and then correcting the times
```
Comment (wholmgren) [inline pvlib/iotools/srml.py]: Add comment like "e.g. the first two rows of hourly data are 100, 200, so interval length = 100"

File: pvlib/iotools/srml.py [hunk 4]
```diff
@@ -134,11 +135,17 @@ def format_index(df):
...
+    # at each former hour. interval_length is determined by taking the
+    # difference of the first two rows of the time column.
+    interval_length = df[df.columns[1]][1] - df[df.columns[1]][0]
+    df_time = df[df.columns[1]] - interval_length
+    if interval_length == 100:
+        # Hourly files do not require fixing the former hour timestamps.
+        times = df_time
+    else:
+        old_hours = df_time % 100 == (100 - interval_length)
```
Comment (wholmgren) [inline pvlib/iotools/srml.py]: this needs a few comments

File: pvlib/iotools/srml.py [hunk 5]
```diff
@@ -134,11 +135,17 @@ def format_index(df):
...
+        times = df_time.where(~old_hours, df_time - 40)
     times = times.apply(lambda x: '{:04.0f}'.format(x))
```
[the remark this answers is not in the document]
Comment (lboeman) [reply pvlib/iotools/srml.py]: Pandas complains about calling .str with an integer after this change: 
```
AttributeError: Can only use .str accessor with string values, which use np.object_ dtype in pandas
```

[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/iotools/srml.py]: ah, right, you've got ints until this line. Probably better to leave as is then.

**Comment by lboeman** [comment]
I updated the logic for fixing the hour times. I didn't think of checking `value % 100 < 60` to locate the indices that were previously aligned with hours, but that logic seems easier to comprehend and slightly cheaper to perform.

**Pull request merged**